client/scripts/views/pages/view_proposal/index.ts

```diff
@@ -54,17 +53,21 @@ const ProposalHeader: m.Component<{
   getSetGlobalReplyStatus: CallableFunction;
   proposal: AnyProposal | OffchainThread;
 }, {
+  isAuthor: boolean;
+  isEditor: boolean;
+  isAdmin: boolean;
   savedEdit: string;
   editing: boolean;
   saving: boolean;
   quillEditorState: any;
   currentText: any;
   topicEditorIsOpen: boolean;
+  editPermissionsIsOpen: boolean;
 }> = {
   view: (vnode) => {
     const { commentCount, proposal, getSetGlobalEditingStatus, getSetGlobalReplyStatus, viewCount } = vnode.attrs;
 
-    const isAdmin = (app.user.isRoleOfCommunity({
+    vnode.state.isAdmin = (app.user.isRoleOfCommunity({
```

Comment:
This seems unintended, why are we assigning to vnode.state every time the view renders?

Reply:
Yes, this is legacy from its place in oninit—let me switch back to a const.